eslint.config.mjs

```diff
@@ -0,0 +1,94 @@
+// @ts-check
+
+import path from 'node:path';
+import {fileURLToPath} from 'node:url';
+import react from '@eslint-react/eslint-plugin';
+import {includeIgnoreFile} from '@eslint/compat';
+import js from '@eslint/js';
+import pluginQuery from '@tanstack/eslint-plugin-query';
+// @ts-expect-error Requires updating tsconfig (see https://github.com/typescript-eslint/typescript-eslint/issues/7284)
+import * as tsParser from '@typescript-eslint/parser';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+const gitignorePath = path.join(
+  path.dirname(fileURLToPath(import.meta.url)),
+  '.gitignore',
+);
+
+const gitExcludePath = path.join(
+  path.dirname(fileURLToPath(import.meta.url)),
+  '.git',
+  'info',
+  'exclude',
+);
+
+const toolingConfig = tseslint.config({
+  name: 'tooling',
+  files: [
+    '*.config.{js,mjs,cjs}',
+    'scripts/*.{js,mjs,cjs}',
+    'expo-config-plugins/*.{js,mjs,cjs}',
+  ],
+  languageOptions: {
+    globals: {
+      ...globals.node,
+      ...globals.nodeBuiltin,
+      ...globals.worker,
+    },
```

Comment:
Should we include `global.jest`? In preparation for #898 ?

Reply:
no, because `globals.jest` is relevant for test files that are run in a jest environment. this configuration is for configuration files related to build tooling.

probably makes sense to add a config in `frontendConfig` that targets test files and specifies the globals there instead